gravitee-am-service/src/main/java/io/gravitee/am/service/impl/UserServiceImpl.java

```diff
@@ -197,6 +197,15 @@ protected UserRepository getUserRepository() {
                     if (enrolledFactors == null || enrolledFactors.isEmpty()) {
                         enrolledFactors = Collections.singletonList(enrolledFactor);
                     } else {
+                        // if current factor is primary, set the others to secondary
+                        if (Boolean.TRUE.equals(enrolledFactor.isPrimary())) {
+                            enrolledFactors = enrolledFactors
```

Comment:
I'd rather use a foreach loop since we are modifying the state of objects in the collection ?